### PR TITLE
Preparations for 3.15.1

### PR DESCRIPTION
**Pull request by algernon**
This PR bumps versions, and adds a NEWS file (in a second commit).

### CODE REVIEW

**Comment by furiel** [comment]
Can you grep for `@version`? There are some exciting places with old versions (for example 3.11), probably we forgot to update those earlier. For example:
```
packaging/debian/syslog-ng.conf
1:@version: 3.11

```

**Comment by pzoleex** [comment]
@kira-syslogng do perftest

File: NEWS.md [hunk 1]
```diff
@@ -1,63 +1,57 @@
...
+ * Support added for `if`/`elif`/`else` blocks to the configuration file syntax.
+   (#1856)
+ * Dramatically improved debug messages during filter/parser evaluation. (#1898)
+ * The `hook-commands` module was added, allowing one to run custom commands on
+   source or destination setup and teardown. (#1951)
+ * Implemented a way to skip processing included config file snippets in case a
+   dependency is missing: The `@requires json` pragma. (#827, #1956)
+ * Basic client-side failover support was implemented. (#1905)
+ * Errors from python destinations are now reported together with any exception
+   text (if any). (#1931)
```
Comment (MrAnno) [inline NEWS.md]: Do we want to include #1911?

File: NEWS.md [hunk 2]
```diff
@@ -1,63 +1,57 @@
...
+ * SDATA block names are now sanitized, in order to not break the spec when we
+   get our SDATA from sources that are more lax (such as JSON). (#1948)
+ * Some internal messages contained key-value pairs where the key had spaces in
+   it, this has been addressed, they do not contain spaces anymore.
+ * Fix compilation on FreeBSD. (#1901)
+ * Fix compilation on SLES 11. (#1897)
+ * Fix compilation on Hurd. (#1912, #1914)
+ * Fix compiltaion on Solaris 10. (#1982, #1983)
+ * Fix compilation on MacOS.
+ * Various compiler warning-related fixes all over the codebase.
```
Comment (MrAnno) [inline NEWS.md]: @algernon Do we want to add #1921 (#1997)?

**Comment by algernon** [comment]
@furiel good catch, thanks! Updated the `@version`s I missed previously, and force-pushed.

**Comment by kira-syslogng** [comment]
success

File: NEWS.md [hunk 3]
```diff
@@ -1,63 +1,57 @@
...
- * Refactor in rewrite() module init (#1818)
- * Missing child program (exit status 127) handling is changed in program() destination:
-    stopping destination instead of polling for the child program (#1817)
- * Refactor in filter() module (#1814)
- * Improve thread synchronization in mainloop and refactor (#1813)
- * Adapted json-c v0.13 API changes to json-parser (#1810)
+ * POSIX RegExp support was dropped from the filters, PCRE remains available. (#1899)
+ * Miscellaneous build-system related fixes and improvements (both autotools and
+   CMake).
+ * Update `lib/json-c` to `json-c-0.13-20171207`. (#1900)
```
Comment (MrAnno) [inline NEWS.md]: I'd add #1932, this is a hude improvement.

File: NEWS.md (hunk 2 above)
Comment (MrAnno) [reply NEWS.md]: I'd also add #1967, #1957.

**Comment by kira-syslogng** [comment]
success

File: NEWS.md (hunk 1 above)
Comment (lbudai) [reply NEWS.md]: @MrAnno: yes, we want

**Comment by pzoleex** [comment]
@kira-syslogng do perftest

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: NEWS.md (hunk 2 above)
Comment (MrAnno) [reply NEWS.md]: #2000 has been merged (fixes #1936).

File: NEWS.md (hunk 1 above)
Comment (algernon) [reply NEWS.md]: ack, added.

Will force-push the branch once I'm done with the rest too.

Comment (MrAnno) [reply NEWS.md]: Thanks :+1:

**Comment by algernon** [comment]
force-pushed the branch, adding news items about all the referenced PRs.

**Comment by kira-syslogng** [comment]
success

**Comment by kira-syslogng** [comment]
Performance test: Build FAILURE, the tests were executed on test branch: master and test suite: functions

**Comment by pzoleex** [comment]
@kira-syslogng do perftest

**Comment by kira-syslogng** [comment]
Performance test: Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Pull request merged**